### ISSUE TITLE
`Nan` logits when performing inference using ModernBERT

### ISSUE THREAD

**Issue by yaswanthg15** [huggingface/transformers#35574]
### System Info

transformers == 4.48.0.dev0
torch == 2.2.2

### Description
I have finetuned ModelBERT model for multi-label task and when performing batched inference I am getting `Nan` values for logits in a batch except for one sample. So, in each batch except one sample all the remaining logits are `Nan`
### Who can help?

@tomaarsen @ArthurZucker 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Simply passing model(**batch) is resulting in this - while with bs=1 no issue exists.
Code: 

```
model.eval()
labels, predictions, confidences = [], [], []

with torch.no_grad():
    for batch in tqdm(val_loader):
        labels.extend(batch['labels'].cpu().numpy())
        batch.pop('labels')
        batch = {k: v.to(device) for k, v in batch.items()}

        outputs = model(**batch)

        probs = torch.sigmoid(outputs.logits)
        preds = (probs > 0.5).int()
        confs = probs

        predictions.extend(preds.cpu().numpy())
        confidences.extend(confs.cpu().numpy())
```

<img width="1124" alt="image" src="https://github.com/user-attachments/assets/fe10bfdb-4921-4733-bae4-7f87d68581f4" />

### Expected behavior

Expecting some values instead of `Nan`

**Comment by tomaarsen** [comment]
Could you add a snippet to reproduce? I've done lots of inference and I've never seen `nan`s yet.

- Tom Aarsen

**Comment by yaswanthg15** [comment]
@tomaarsen While it is not possible for me to make the model public - but coming to the issue I am performing a simple inference using pytorch loop.

**Comment by ArthurZucker** [comment]
In order to help we would need to know: 
- the dtype
- the attention implementation 
This should help us as a lot of nans can come from, sdpa or casting

**Comment by yaswanthg15** [comment]
IDK - it seems random as today I am the getting correct logits without any change.

**Comment by saran9991** [comment]
@yaswanthg15 Did you figure out what the problem is? I'm facing the same issue.

**Comment by tomaarsen** [comment]
I've seen `nan`'s more often now, and often times they were resolved by upgrading `torch`:
```
pip install -U torch
```
(the command might be a bit different depending on whether you want CPU or CUDA-compatible torch)

- Tom Aarsen
